### ISSUE TITLE
gatsby-source-wordpress fails to build development when sourcing from Local via https:// on GatsbyV5

### ISSUE THREAD

**Issue by CloudBop** [gatsbyjs/gatsby#37417]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I get the following build error when trying to source WP from Local via HTTPS. The plugin fails before any gatsby-source-wordpress logs to the console
``` ERROR #gatsby-source-wordpress_111002  PLUGIN

 gatsby-source-wordpress  Error: connect ECONNREFUSED ::1:443
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1300:16)

GraphQL request to https://wpgqlgatsbyv5.local/graphql failed.

Please ensure the following statements are true
  - your WordPress URL is correct in gatsby-config.js
  - your server is responding to requests
  - WPGraphQL and WPGatsby are installed and active in your WordPress backend
  - Your WordPress debug.log does not contain critical errors

 ```
The error doesn't occur if I switch the URL string in gatsby-plugins.gatsby-source-wordpress options.url to HTTP:// - there is only a Gatsby warning.
```
warn  gatsby-source-wordpress 

The Url set in plugin options has a different protocol than the Url saved in WordPress general settings.

options.url: http://wpgqlgatsbyv5.local/graphql
WordPress settings: https://wpgqlgatsbyv5.local

This may cause subtle bugs, or it may be fine.
Please consider addressing this issue by changing your WordPress settings or plugin options accordingly.
```

This leads me to believe that the error isn't due to the WP installation (LocalByFlywheel) but something to do with gatsby-plugin-source build process. 

Using Gatsbyv4, I remember using the ENV flag to get around SSL development.build errors.

`NODE_TLS_REJECT_UNAUTHORIZED=0`

Perhaps GatsbyV5 has dropped support for this flag? I've tried using it and can see the warning printed during development.build process. Unfortunately it doesn't seem to fix this issue.
```
 ERROR  UNKNOWN

(node:18589) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by
disabling certificate verification.
(Use `node --trace-warnings ...` to show where the warning was created)
```

Curiously, ```Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED``` message doesn't appear when using the WP_URL as http://

This is all further complicated because Local requires SSL to be configured via macOS keychain. I am pretty sure that I have everything configured properly as the development-site.local looks to be satisfying browsers.

### Reproduction Link

https://github.com/CloudBop/gatsbyv5-wpgql-from-starter-wordpress-blog

### Steps to Reproduce

1. Create a new WP site using local 
2. Configure the SSL via Local, WP->reading setting and macOS->Keychain.
3. Install WPGraphQL and WpGatsby.
4. Install the Gatsby repo
5. Ensure that the correct URL is set in ```gatsby-config.js```
6. Run ```gatsby develop  ```

### Expected Result

The **gatsby-source-wordpress** plugin starts the build process.

### Actual Result

Development build fails early in build process, see information below. 

```
 gatsby develop

success compile gatsby files - 0.491s
success load gatsby config - 0.035s
success load plugins - 1.236s
warn gatsby-plugin-react-helmet: Gatsby now has built-in support for modifying the document head. Learn more at https://gatsby.dev/gatsby-head
success onPreInit - 0.014s
success initialize cache - 0.225s
success copy gatsby files - 0.224s
success Compiling Gatsby Functions - 0.278s
success onPreBootstrap - 0.304s

 ERROR  UNKNOWN

(node:17693) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by
disabling certificate verification.
(Use `node --trace-warnings ...` to show where the warning was created)


 ERROR #gatsby-source-wordpress_111002  PLUGIN

 gatsby-source-wordpress  Error: connect ECONNREFUSED ::1:443
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1300:16)

GraphQL request to https://wpgqlgatsbyv5.local/graphql failed.

Please ensure the following statements are true
  - your WordPress URL is correct in gatsby-config.js
  - your server is responding to requests
  - WPGraphQL and WPGatsby are installed and active in your WordPress backend
  - Your WordPress debug.log does not contain critical errors

not finished createSchemaCustomization - 0.230s
not finished  gatsby-source-wordpress  ensuring plugin requirements are met - 0.222s
```

### Environment

```shell
System:
    OS: macOS 12.6.2
    CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v18.12.1/bin/npm
  Languages:
    Python: 3.7.4 - /Users/somebody/opt/anaconda3/bin/python
  Browsers:
    Chrome: 108.0.5359.124
    Firefox: 108.0.2
    Safari: 16.2
  npmPackages:
    gatsby: ~5.3.2 => 5.3.2 
    gatsby-plugin-image: ^3.3.2 => 3.3.2 
    gatsby-plugin-manifest: ^5.3.1 => 5.3.1 
    gatsby-plugin-offline: ^6.3.1 => 6.3.1 
    gatsby-plugin-react-helmet: ^6.3.0 => 6.3.0 
    gatsby-plugin-sharp: ^5.3.2 => 5.3.2 
    gatsby-source-filesystem: ^5.3.1 => 5.3.1 
    gatsby-source-wordpress: ^7.3.2 => 7.3.2 
    gatsby-transformer-sharp: ^5.3.1 => 5.3.1
```

### Config Flags

No config flags, but did use ENV_VAR in Gatsby@V4

``` NODE_TLS_REJECT_UNAUTHORIZED=0 ```

**Comment by tyhopp** [comment]
Hi @CloudBop, can you reproduce this minimally when Local is removed from the equation? That would help confirm whether this is an issue with `gatsby-source-wordpress` or not.

**Comment by CloudBop** [comment]
Hi @tyhopp , thanks for getting back to me.

I thought I had resolved this issue by using the ``` gatsby develop --https``` but as soon as I cleared the cache and tried building from another Local site it failed again. Leaving me questioning whether I had actually correctly updated the protocol on the URL string. I'm like 90% sure I did have a couple of successful builds before they started failing again.

The site builds from the live WP url, I haven't tried setting up via another local WP setup.  I'll try setting up via Docker this weekend. 

I have a feeling it might be a Monterey (tried disabling firewall to see if something was blocking 433) /Local issue, I do indeed see people having SSL issues on the Local forums, but it does seem to be behaving when I setup the SSL via Keychain and go to the page in a browser. 

Frustratingly, development works from the exact same Local sites if I revert to my site back to using GatsbyV4. This requires no change with Local/WP. I suppose this may be because I am also reverting back to node14.

Interestingly if I run the following from a browser dev tools, the https:// protocol will work, whereas the http:// protocol will error. 

This seems to be the opposite of what's happening when I try and boot up ```gatsby development``` If the protocol is https:// then things fail. If it's http:// although there is a warning, the site still builds.

```
fetch(`https://mysite.local/graphql`, {
	method: 'POST',
	headers: {
		'Content-Type': 'application/json'
	},
	body: JSON.stringify({
		query: `
									{
											generalSettings {
													url
													
											}
									}
							`,
	}),
})
	.then(res => res.json())
	.then(res => {
		console.log('res', res)
	}).catch(e => {
		console.log('e', e)
	})
```

It does seem to be failing on the wpgraphql, I've updated the plugin debug settings and this is what get spat back out. I've thrown a couple of logs in my local version of gatsby-source-wordpress to see if I could diagnose any further but it's just the same error the plugin throws.

```
success  gatsby-source-wordpress  step -> setRequestHeaders - 0.001s
success initialize cache - 0.221s
success copy gatsby files - 0.163s
success Compiling Gatsby Functions - 0.248s
success onPreBootstrap - 0.304s
verbose Creating 3 worker
verbose Transition to "initializingData"
success  gatsby-source-wordpress  step -> setGatsbyApiToState - 0.006s

CHECKME - ensurePluginRequirementsAreMet LN 283

e FetchError: request to https://wpgqlgatsbyv5.local/graphql failed, reason: connect ECONNREFUSED ::1:443
    at ClientRequest.<anonymous>
(/Volumes/Storage/Projects/gatsby-sites/gatsbyV5-starter-wordpress-blog/node_modules/node-fetch/lib/index.js:1491:11)
    at ClientRequest.emit (node:events:513:28)
    at TLSSocket.socketErrorListener (node:_http_client:494:9)
    at TLSSocket.emit (node:events:513:28)
    at emitErrorNT (node:internal/streams/destroy:151:8)
    at emitErrorCloseNT (node:internal/streams/destroy:116:3)
    at processTicksAndRejections (node:internal/process/task_queues:82:21) {
  type: 'system',
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED'
}
firstcatch blankGetRequest

 ERROR  UNKNOWN

 gatsby-source-wordpress  GraphQL query: {
  isWpGatsby
}



 ERROR  UNKNOWN

 gatsby-source-wordpress  GraphQL query: {
  generalSettings {
    url
  }
  wpGatsby {
    arePrettyPermalinksEnabled
  }
}


info  gatsby-source-wordpress  GraphQL vars: {"wpgqlVersion":">=1.1.2 <2.0.0","wpgatsbyVersion":">=0.9.0 <3.0.0"}

 ERROR  UNKNOWN

 gatsby-source-wordpress  GraphQL query: query WPGatsbyCompatibility($wpgqlVersion: String!, $wpgatsbyVersion: String!) {
  wpGatsbyCompatibility(
    wpGatsbyVersionRange: $wpgatsbyVersion
    wpGQLVersionRange: $wpgqlVersion
  ) {
    satisfies {
      wpGQL
      wpGatsby
    }
  }
}



 ERROR #gatsby-source-wordpress_111002  PLUGIN

 gatsby-source-wordpress  Error: connect ECONNREFUSED ::1:443
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1300:16)

GraphQL request to https://wpgqlgatsbyv5.local/graphql failed.

Please ensure the following statements are true
  - your WordPress URL is correct in gatsby-config.js
  - your server is responding to requests
  - WPGraphQL and WPGatsby are installed and active in your WordPress backend
  - Your WordPress debug.log does not contain critical errors

not finished createSchemaCustomization - 0.304s
not finished  gatsby-source-wordpress  step -> ensurePluginRequirementsAreMet - 0.283s
not finished  gatsby-source-wordpress  ensuring plugin requirements are met - 0.278s
```

It's a really odd issue...

**Comment by hazuremon** [comment]
I am having a similar issue with development build sourcing from Local, the different is that it's just a http://localhost:[port #] and not https. It has the same error message as shown by @CloudBop.

Tested that the GraphQL URL, http://localhost:[port#]/graphql is working, and a production build is also working.

I did manage to get past the error by switching the source from localhost and 127.0.0.1, and run a gatsby clean before hand.

On latest version of gatsby and gatsby-source-wordpress, 7.4.0.
